sycl/include/sycl/ext/oneapi/matrix/matrix.hpp

```diff
@@ -16,16 +16,14 @@
 
 #include <sycl/feature_test.hpp>
 
-// the default is matrix-jit-use but existing tests in llvm-test-suite won't
-// fail because we have the "unnecessary" use value
-#if (SYCL_EXT_ONEAPI_MATRIX == 1)
-#include <sycl/ext/oneapi/matrix/matrix-jit.hpp>
-#include <sycl/ext/oneapi/matrix/static-query.hpp>
+#if (SYCL_EXT_ONEAPI_MATRIX == 3)
```

Comment:
@JackAKirk it doesn't look like a correct usage of `SYCL_EXT_ONEAPI_MATRIX` (users shouldn't pass such macro during compilation, see https://github.com/intel/llvm/pull/6662#issuecomment-1252439065 ). What would be an appropriate replacement for it?

Reply:
There isn't an intention of using any macro in the future in order to decide the APIs to use. Initially for the initial Tensor Cores experimental implementation we followed the existing pattern used by the AOT and JIT intel implementations by setting a new value to the feature test macro.  The SYCL spec itself doesn't appear to forbid this, although it sounds like the extension document itself contradicts the implementation in this regard!

We could add a temporary separate macro analogous to SYCL_EXT_ONEAPI_MATRIX_LEGACY_API if it is considered very important. I don't mind. SYCL_EXT_ONEAPI_MATRIX_LEGACY_CUDA_API for example.
Since the interfaces for the combined Intel/CUDA implementation has been decided I could also open a PR that supports both Intel and CUDA backends within the same implementation. https://github.com/intel/llvm/pull/5920 is blocked on a spirv builtin atm, so it sounds like we shouldn't wait on this PR before starting the combined implementation.

Reply:
I just talked with @dkhaldi, and it sounds like she prefers to have a different name for the macro that chooses the API.